pkg/pki/pki_manager.go

```diff
@@ -0,0 +1,87 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pki
+
+import (
+	"context"
+	"crypto/tls"
+
+	"github.com/banzaicloud/kafka-operator/api/v1alpha1"
+	"github.com/banzaicloud/kafka-operator/api/v1beta1"
+	"github.com/banzaicloud/kafka-operator/pkg/pki/certmanagerpki"
+	"github.com/banzaicloud/kafka-operator/pkg/pki/vaultpki"
+	pkicommon "github.com/banzaicloud/kafka-operator/pkg/util/pki"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// MockBackend is used for mocking during testing
+var MockBackend = v1beta1.PKIBackend("mock")
+
+// GetPKIManager returns a PKI/User manager interface for a given cluster
+func GetPKIManager(client client.Client, cluster *v1beta1.KafkaCluster) pkicommon.PKIManager {
+	switch cluster.Spec.ListenersConfig.SSLSecrets.PKIBackend {
```

Comment:
If SSL isn't specified in KafkaCluster CR this will be a nil pointer dereference when the `GetPKIManager` function will be called in `checkFinalizers`.